### PR TITLE
HDDS-6824. Revert changes made in proto.lock by HDDS-6768.

### PR DESCRIPTION
**Pull request by nandakumar131**
## What changes were proposed in this pull request?
Revert changes made in proto.lock by HDDS-6768

## What is the link to the Apache JIRA
HDDS-6824

## How was this patch tested?
Existing unit tests.

### CODE REVIEW

File: hadoop-ozone/interface-client/src/main/proto/OmClientProtocol.proto [hunk 1]
```diff
@@ -101,7 +101,8 @@ enum Type {
 
   RevokeS3Secret = 93;
 
-  PurgeDirectories = 94;
+  PurgePaths = 94; // [deprecated=true]
```
Comment (rakeshadr) [inline hadoop-ozone/interface-client/src/main/proto/OmClientProtocol.proto:104]: Any specific reason for commented out the statement. Should it be 

`PurgePaths = 94  [deprecated=true];`

File: hadoop-ozone/interface-client/src/main/proto/OmClientProtocol.proto [hunk 2]
```diff
@@ -201,7 +202,8 @@ message OMRequest {
 
   optional RevokeS3SecretRequest            RevokeS3SecretRequest          = 93;
 
-  optional PurgeDirectoriesRequest          purgeDirectoriesRequest        = 94;
+  optional PurgePathsRequest                purgePathsRequest              = 94; // [deprecated = true]
```
Comment (rakeshadr) [inline hadoop-ozone/interface-client/src/main/proto/OmClientProtocol.proto:205]: Here again the same as above. Should it be without comment line ?

File: hadoop-ozone/interface-client/src/main/proto/OmClientProtocol.proto [hunk 3]
```diff
@@ -297,7 +299,8 @@ message OMResponse {
 
   optional ListTrashResponse                  listTrashResponse            = 91;
   optional RecoverTrashResponse               RecoverTrashResponse         = 92;
-  optional PurgeDirectoriesResponse           purgeDirectoriesResponse     = 93;
+  optional PurgePathsResponse                 purgePathsResponse           = 93; // [deprecated = true]
```
Comment (rakeshadr) [inline hadoop-ozone/interface-client/src/main/proto/OmClientProtocol.proto:302]: Here again the same as above. Should it be without comment line ?

I could see everywhere the deprecated line is commented out.

**Review by rakeshadr — COMMENTED** [review]
Thanks @nandakumar131 for the quick response to handle the backward incompatibility case.

**Review by mukul1987 — APPROVED** [review]
+1, LGTM

File: hadoop-ozone/interface-client/src/main/proto/OmClientProtocol.proto (hunk 1 above)
Comment (nandakumar131) [reply hadoop-ozone/interface-client/src/main/proto/OmClientProtocol.proto]: deprecated keyword is not supported for enum in protobuf 2.5

File: hadoop-ozone/interface-client/src/main/proto/OmClientProtocol.proto (hunk 3 above)
Comment (nandakumar131) [reply hadoop-ozone/interface-client/src/main/proto/OmClientProtocol.proto]: Fixed it

File: hadoop-ozone/interface-client/src/main/proto/OmClientProtocol.proto [hunk 4]
```diff
@@ -101,7 +101,8 @@ enum Type {
 
   RevokeS3Secret = 93;
 
-  PurgeDirectories = 94;
+  PurgePaths = 94; // [deprecated = true]
```
Comment (rakeshadr) [inline hadoop-ozone/interface-client/src/main/proto/OmClientProtocol.proto:104]: @nandakumar131 I think, below test case will be happy, if we revert `case PurgePaths:` in OmUtils.java.

```
Error:  org.apache.hadoop.ozone.om.ratis.TestOzoneManagerRatisServer.testIsReadOnlyCapturesAllCmdTypeEnums  Time elapsed: 0.693 s  <<< FAILURE!
java.lang.AssertionError: PurgePaths is not categorized in OmUtils#isReadyOnly
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.junit.Assert.assertFalse(Assert.java:65)
	at org.apache.hadoop.ozone.om.ratis.TestOzoneManagerRatisServer.testIsReadOnlyCapturesAllCmdTypeEnums(TestOzoneManagerRatisServer.java:194)
```

**Review by rakeshadr — APPROVED** [review]
+1 LGTM, thanks @nandakumar131 for the contribution.

Pending one comment to fix the test case failure.

File: hadoop-ozone/interface-client/src/main/proto/OmClientProtocol.proto (hunk 4 above)
Comment (nandakumar131) [reply hadoop-ozone/interface-client/src/main/proto/OmClientProtocol.proto:104]: Fixed it. Thanks @rakeshadr.

**Pull request merged**